rng.h

```diff
@@ -58,9 +58,9 @@ class Xoroshiro128Random
     }
     int nextInt(int n) {
         if (n <=0) return 0;
-        int r =  (int)((n & -n) == n ? nextLong() & n - 1 // power of two
+        int r =  (int)((n & -n) == n ? (nextLong() & n) - 1 // power of two
                        : (unsigned long long)(((unsigned long long)nextLong() >> 32) * n) >> 32);
-        return r;
+        return r < 0 ? 0 : r;
```

Comment:
@lixun910 Here the algorithm will return -1 in some cases, which is not correct. I need to check why this happens. But this should fix the issues that some functions calling `nextInt()` crashes rgeoda, see https://github.com/GeoDaCenter/rgeoda/issues/39 https://github.com/GeoDaCenter/rgeoda/issues/38 https://github.com/GeoDaCenter/rgeoda/issues/33